packages/palette/src/elements/TimeRemaining/TimeRemaining.ios.tsx

```diff
@@ -0,0 +1,77 @@
+import { DateTime, Duration } from "luxon"
+import React, { useState } from "react"
+import { Flex, Sans } from "../"
+import { color } from "../../helpers"
+import { useInterval } from "../../utils/useInterval"
+
+function padWithZero(num: number) {
+  return num.toString().padStart(2, "0")
+}
+
+const SEPARATOR = <>&nbsp;&nbsp;</>
+
+/** TimeRemaining */
+export const TimeRemaining: React.SFC<{
+  endDate: string
+  labelWithTimeRemaining?: string
+  labelWithoutTimeRemaining?: string
+  timeEndedDisplayText?: string
+  trailingText?: string
+  currentTime?: DateTime
+  highlight: Parameters<typeof color>[0]
+}> = ({
+  endDate,
+  highlight = "purple100",
+  labelWithTimeRemaining,
+  labelWithoutTimeRemaining,
+  timeEndedDisplayText,
+  trailingText,
+  currentTime,
+}) => {
+  const now = currentTime ? currentTime : DateTime.local()
+  const [duration, setDuration] = useState(
+    Duration.fromISO(
+      DateTime.fromISO(endDate)
+        .diff(now)
+        .toString()
+    )
+  )
+
+  useInterval(() => {
```

Comment:
Why `useInterval` vs `setInterval`?

Reply:
You mean why is it named that way or why don't I just use a regular setInterval?

Reply:
> why don't I just use a regular setInterval
 ^

Reply:
the `useInterval` hook provides a state object that gets set every second which refreshes the render. Using a regular `setInterval` I'd need to add state to the component, the hook handles the render update and also cleans up the interval on unmount. Best in my opinion for one function to handle everything and keeps us from adding state, hooks and duplicating code.

Reply:
nice 👍